cpp/src/parquet/arrow/reader.cc

```diff
@@ -114,6 +116,32 @@ class ColumnReaderImpl : public ColumnReader {
     return Status::OK();
   }
 
+  ::arrow::Result<std::shared_ptr<::arrow::ChunkedArray>> NextBatch(
+      int64_t batch_size) final {
+    std::shared_ptr<::arrow::ChunkedArray> out;
+    RETURN_NOT_OK(NextBatch(batch_size, &out));
+    return out;
+  }
+
+  Future<std::shared_ptr<ChunkedArray>> NextBatchAsync(
+      int64_t batch_size, ::arrow::internal::Executor* io_executor,
+      ::arrow::internal::Executor* cpu_executor) final {
+    Future<> load_fut = ::arrow::DeferNotOk(
+        io_executor->Submit([this, batch_size] { return LoadBatch(batch_size); }));
```

Comment:
Would it matter that `io_executor` would consume some CPU?

`LeafReader::LoadBatch` might decompress page, decode records, parse def-rep levels. They're all cpu-intensive.

Reply:
Yes I see it in comment, just saying that it might harm. Perfect IO might be so tricky and need to wrap `RangeCache` and `RandomAccessFile`. This looks good to me

Reply:
Another point is that seems that `LoadBatch` would only depends on sync-io api? So it would not cause deadlock when `LoadBatch` waiting for page io? (which could happen when `use_thread` with previous scan?)